src/main/python/integratedPython.ts

```diff
@@ -61,40 +67,49 @@ export const getIntegratedPython = async (
     onProgress: (percentage: number, stage: 'download' | 'extract') => void
 ): Promise<PythonInfo> => {
     const platform = getPlatform();
-    const pythonPath = path.resolve(path.join(directory, getExecutableRelativePath(platform)));
+    const { url, version, path: relativePath } = downloads[platform];
 
+    const pythonPath = path.resolve(path.join(directory, relativePath));
     const pythonBinExists = await checkFileExists(pythonPath);
 
-    if (!pythonBinExists) {
-        log.info(`Integrated Python not found at ${pythonPath}`);
+    if (pythonBinExists) {
+        const pythonInfo = await checkPythonPaths([pythonPath]);
+        if (eq(pythonInfo.version, version)) {
+            return pythonInfo;
+        }
+        // Invalid version, remove legacy integrated python
+        const legacyPythonFolder = path.resolve(path.join(directory, '/python'));
+        await fs.rm(legacyPythonFolder, { recursive: true, force: true });
+    }
```

Comment:
The `/python` folder might exist even if the python bin doesn't. So you always need to do the `rm -rf`.